docs/src/components/mdx/code-block.tsx

```diff
@@ -1,22 +1,80 @@
 import Button from '@pluralsight/ps-design-system-button'
+import Dropdown from '@pluralsight/ps-design-system-dropdown'
 import Theme, { useTheme } from '@pluralsight/ps-design-system-theme'
 import cx from 'classnames'
 import Prism from 'prismjs/components/prism-core'
 import Highlight, { PrismTheme, defaultProps } from 'prism-react-renderer'
 import React, { HTMLAttributes, useEffect, useState } from 'react'
 import { CopyToClipboard } from 'react-copy-to-clipboard'
 
+import { H2 } from '../mdx'
 import styles from './code-block.module.css'
 import { darkTheme, lightTheme } from './code-block-theme'
 
 interface CodeBlockProps extends HTMLAttributes<HTMLDivElement> {
+  metastring?: null
   live?: boolean
 }
 
 export const CodeBlock: React.FC<CodeBlockProps> = (props) => {
-  const { children: code } = props
+  const isSwitcher = /switcher/.test(props.metastring)
   const language = props.className.replace(/language-/, '')
 
+  const codes = props.children.split('\n\n---\n\n')
+  const options = isSwitcher
+    ? Array.from(Array(codes.length)).map((_, i) => ({
+        value: i,
+        label: 'Example #' + i,
+      }))
+    : [{ value: 0, label: 'Single example' }]
+
+  const [selectedOption, setSelectedOption] = React.useState(options[0].value)
```

Comment:
i'm not getting the initially selected option in the dropdown value

![Screen Shot 2020-09-15 at 9 37 24 AM](https://user-images.githubusercontent.com/226356/93232380-14a76c80-f737-11ea-8bb7-63a437b1284d.png)